packages/x-tree-view/src/RichTreeView/RichTreeView.tsx

```diff
@@ -172,6 +172,7 @@ RichTreeView.propTypes = {
   experimentalFeatures: PropTypes.shape({
     indentationAtItemLevel: PropTypes.bool,
     labelEditing: PropTypes.bool,
+    lazyLoading: PropTypes.bool,
```

Comment:
I don't understand this one. If lazing loading can be detected with the prop. What's the advantage of this? We could be doing: 

```suggestion
```

and then: `unstable_treeViewDataSource`, `unstable_treeViewDataSourceCache`. I thought the point of `experimentalFeatures` is for things that we can't change with a prop.

I have the same confusion with Next.js: https://nextjs.org/docs/app/building-your-application/upgrading/version-15#serverexternalpackages, I don't understand why they handle this this way.

Reply:
We went for consistency here, each new major feature is hidden under an `experimentalFeature`.
For me this has two main benefits:

1. If you have several props, you don't need to prefix each one of them (plus the API methods, plus the utility hooks etc...)
2. Users don't have a breaking change when the feature becomes stable. They can remove there `experimentalFeature` but there app will work fine even if they keep it. With the `unstable_` prefix, there is a breaking change that we have to care about (by exporting two version with and without the prefix for some time) and that is still a cost for our users.
3. Some feature have to be flagged as `experimentalFeatures` because as you said we can't do it with props (because they are opt-out, like the aggregation). So I prefer to be consistent and not have the user learn a new way of enabling an experimental feature for each new feature.

Reply:
@flaviendelangle The limits I see:

1. It's not clear what's stable and what's unstable. This would be clearer

```jsx
<TreeView
 experimentalFeatures={{
    treeViewDataSource: 
    treeViewDataSourceCache:
  }}
```

2. The terminology is confusing, we call the rest unstable. "experimental" sounds likely that it will get dropped, "unstable" feels like a higher degree of commitment. 

```jsx
<TreeView
 unstableFeatures={{
    treeViewDataSource: 
    treeViewDataSourceCache:
  }}
```

Reply:
> The terminology is confusing, we call the rest unstable. "experimental" sounds likely that it could get dropped, "unstable" feels like a higher degree of commitment.

On the other hand, our assumption was that "unstable" was more repelling than "experimental" which is why we called this prop "experimentalFeatures" instead of "unstableFeatures" on the grid.

> The API could change when it becomes stable though.

But most of the time it doesn't on the experimental features we had in the past.

Reply:
@oliviertassinari 

> It's not clear what's stable and what's unstable. This would be clearer
```<TreeView
  experimentalFeatures={{
    treeViewDataSource: 
    treeViewDataSourceCache:
  }}```
  
  I agree, we should add both props that are introduced to the experimentalFeatures list (or unstableFeatures if we decide to rename) 👌. 
 We weren't sure about the exact dx we wanted here and how similar it should be to the one the grid has when I flagged this as experimental (did it right at the beginning of my exploration). So the inconsistency between what's in experimental features and the names of the props is due to the fact that this is still WIP. 🙈

Reply:
If we think `experimentalFeatures={{ treeViewDataSource: true  }}` is not clear enough to tell that the cache is experimental as well, I'm fine splitting the two.
I felt like both were close enough to keep them together (which is what the grid did I think) but that's debatable.

Reply:
React uses both, unstable and experimental https://github.com/facebook/react/blob/main/packages/react/index.js#L63. Confusing perceptions in https://www.linuxquestions.org/questions/debian-26/unstable-vs-experimental-331158/.

No a strong view, but unless there are major cons with this, going like this seems good enough:

- stick to a single term in code, and in docs, it feels unlikely that we can't reliability communicate unstable > experimental or unstable < experimental.
- we could introduce a pre GA prefix, effective what we use unstable_ so that we can then use unstable for really unstable things like AI integration. But so we really need that level of complexity? Until we have too many examples where critical feedback isn't left on the features during the non stable phase and is only left after, please no, let's be lazy.
- with a single term, use unstable because it's shorter, a bit closer to what React does?
- prefer to flatten unstable props rather than grouping them, our API docs generation struggles in documenting nested props. Not sure about the value of grouping in the first place, don't see why Next.js group them
- have deprecations when promoting a unstable API to stable.
- make all components follow the same policy, Base UI, etc.